src/module.rs

```diff
@@ -1073,6 +1092,176 @@ impl DataLayout {
+            Type::FPType(FPType::Half) => Some(TypeSize::fixed(16)),
+            #[cfg(feature = "llvm-11-or-greater")]
+            Type::FPType(FPType::BFloat) => Some(TypeSize::fixed(16)),
+            Type::FPType(FPType::Single) => Some(TypeSize::fixed(32)),
+            Type::FPType(FPType::Double) => Some(TypeSize::fixed(64)),
+            Type::FPType(FPType::FP128) => Some(TypeSize::fixed(128)),
+            Type::FPType(FPType::X86_FP80) => Some(TypeSize::fixed(80)),
+            Type::FPType(FPType::PPC_FP128) => Some(TypeSize::fixed(128)),
+            Type::VectorType {
+                element_type,
+                num_elements,
+                scalable,
+            } => self
+                .get_type_size_in_bits(types, element_type)
+                .map(|elt_sz| {
+                    TypeSize::new(
+                        u64::try_from(*num_elements).unwrap() * elt_sz.quantity,
+                        *scalable,
+                    )
+                }),
+            Type::ArrayType {
+                element_type,
+                num_elements,
+            } => self
+                .get_type_size_in_bits(types, element_type)
+                .map(|elt_sz| {
+                    TypeSize::fixed(u64::try_from(*num_elements).unwrap() * elt_sz.quantity)
+                }),
+            Type::StructType {
+                element_types,
+                is_packed,
+            } => {
+                let mut sz = 0;
+                for elt_ty in element_types {
+                    let align = if *is_packed {
+                        Alignment { abi: 1, pref: 1 }
+                    } else {
+                        self.alignments.type_alignment(elt_ty).clone()
+                    };
+                    if !is_aligned_abi(&align, sz) {
+                        sz = align_to_abi(sz, &align);
+                    }
+                    if let Some(elt_sz) = self.get_type_alloc_size_in_bits(types, elt_ty) {
+                        sz += elt_sz.to_fixed();
+                    } else {
+                        return None;
+                    }
+                }
+                Some(TypeSize::fixed(sz))
+            },
+            Type::NamedStructType { name } => {
+                if let Some(NamedStructDef::Defined(def)) = types.named_struct_def(name) {
+                    self.get_type_size_in_bits(types, &def)
+                } else {
+                    None
+                }
+            },
+            Type::X86_AMXType => Some(TypeSize::fixed(8192)),
+            Type::X86_MMXType => Some(TypeSize::fixed(64)),
+            Type::FuncType { .. } => None,
+            Type::LabelType => None,
+            Type::MetadataType => None,
+            Type::TokenType => None,
+            Type::VoidType => None,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct TypeSize {
+    quantity: u64,
```

Comment:
instead of `quantity`, can we be explicit about whether this is `bits` or `bytes`?